openmc/tallies.py

```diff
@@ -762,6 +762,8 @@ def _can_merge_filters(self, other):
 
                 # If filters are the second mergeable filters encountered
                 elif filter1.can_merge(filter2) and merge_filters:
+                    merge_filters = True
+                    mergeable_filter = True
```

Comment:
This was added when I was toying around with a fix and accidentally was retained in my commit.